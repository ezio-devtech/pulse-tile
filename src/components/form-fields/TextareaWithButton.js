import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

export default class TextareaWithButton extends PureComponent {
  static propTypes = {
    button: PropTypes.object.isRequired,
    fieldProps: PropTypes.object.isRequired,
  };

  render() {
    console.log(this.props.fieldProps);
    return (
      <div>
        <div className="control-group right buttoned-control-group">
          {this.props.button}
        </div>
        <Field
          {...this.props.fieldProps}
        />
      </div>
    )
  }
}

