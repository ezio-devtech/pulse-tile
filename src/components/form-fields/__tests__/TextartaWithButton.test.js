import React from 'react';
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

import TextareaWithButton from '../TextareaWithButton';

// Was removed from the main repository for extracting DiaryEntry plugin
// import {valuesLabels, valuesNames} from '../../pages/DiaryEntry/forms.config';

const testProps = {
  button: <button>test</button>,
  fieldProps: {
    // label:  valuesLabels.TYPE,
    // name:   valuesNames.TYPE,
    // id:     valuesNames.TYPE,
  },
};

describe('Component <TextareaWithButton />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <TextareaWithButton
        {...testProps}
      >
      </TextareaWithButton>);

    expect(component.find('.buttoned-control-group')).toHaveLength(1);
    expect(component.find('Field')).toHaveLength(1);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
