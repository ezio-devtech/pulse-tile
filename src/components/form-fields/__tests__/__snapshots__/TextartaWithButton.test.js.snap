// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <TextareaWithButton /> should renders with all props correctly 1`] = `
<div>
  <div
    className="control-group right buttoned-control-group"
  >
    <button>
      test
    </button>
  </div>
  <Field />
</div>
`;
